de.metas.procurement.base/src/main/java/de/metas/procurement/base/order/interceptor/PMM_PurchaseCandidate_OrderLine.java

```diff
@@ -95,19 +96,22 @@ public static PMMBalanceChangeEvent createPMMBalanceChangeEvent(final I_PMM_Purc
 		final I_PMM_PurchaseCandidate candidate = alloc.getPMM_PurchaseCandidate();
 		final BigDecimal qtyOrdered = isReversal ? alloc.getQtyOrdered().negate() : alloc.getQtyOrdered();
 		final BigDecimal qtyOrderedTU = isReversal ? alloc.getQtyOrdered_TU().negate() : alloc.getQtyOrdered_TU();
+
+		final I_M_HU_PI_Item_Product huPIItemProduct = Services.get(IPMMPurchaseCandidateBL.class).getM_HU_PI_Item_Product_Effective(candidate);
+
```

Comment:
Pls introduce and use IPMMPurchaseCandidateBL.getM_HU_PI_Item_Product_Effective_ID(model) to avoid unnecessary database loads